Update Checkbox.md (#2391)
coveo/plasma: packages/website/docs/Checkbox.md

[Description]: # 

> [Checkboxes allow the selection of multiples options from a list. Checkboxes can also be used to turn one option on or off.]: # 

Reviewer: What do you use these 2 lines for?

Author: It's for the description that will appear on the page header and tiles in the home page. They are not yet integrated.

> # Best Practices 

Reviewer: We are in the `Guidelines` tab, I don't think a `Best Practices` title is needed.

![image](https://user-images.githubusercontent.com/966550/150802515-97d20ac2-98e0-42ba-a173-961d9cc61e97.png)

Author: We could consider changing that in phase 1, but right now, they expose some bit-sized guidelines that could quickly give you a confirmation weither this input is or is not the right one for your use case.

Author: We could consider changing that in phase 1, but right now, they expose some bit-sized guidelines that could quickly give you a confirmation if this input is or is not the right one for your use case.

# Best Practices

Use checkboxes to have users select one or more options from a list, or none at all.

Use a stand-alone checkbox to toggle a single option on/off if the result is not immediately visible. Conversely, if the result applies immediately, for example a change of display, use a [Toggle](#/form/FlatSelect) instead.

-   E.g. if a final confirmation is required to apply the effect.

Aim for seven or less options. If that's impossible, consider breaking your set of options into several smaller logical sets or using the [Multi select](#/form/MultiSelect) instead.

> Options should be sorted by most relevant unless a more suited ordering rationale applies, for example when listing size or security level options. If no rationale stands out, place options in alphanumerical order.

Reviewer 2: ```suggestion
Options should be listed by most relevant unless a more suited ordering rationale applies, for example when listing size or security level options. If no rationale stands out, place options in alphanumerical order.
```
Just for coherence.

## Labeling

Keep titles and labels short, preferably **under three words**.

### Title

Always use a title when offering multiple options.

Provide a descriptive title without action verbs. For example, write "Grocery items" rather than "Select all grocery items".

For a stand-alone checkbox, the title is optional. Use a title only if you need to dissociate the stand-alone checkbox from other adjacent checkboxes.

### Labels

Labels identify each option and should be self-explanatory.

Use a consistent writing style for all options in the list.

For a stand-alone checkbox, use a positive and active wording that clearly indicates what happens if the checkbox is selected.

> -   For example, write "Send me a text notification" rather than "Email notification".

Reviewer 2: ```suggestion
For example, write "Send me a text notification" rather than "Email notification".
```
Just removing the lone bullet point :)

## Help Text and Instructions

> Instructions provide more context on the outcome of the choice at-hand. Use it sparingly, only when it is critical to the user's understanding.

Reviewer 2: ```suggestion
Instructions provide more context on the outcome of the choice to make. Use it sparingly, only when it is critical to the user's understanding.
```

## Feedback and Validation

If there is a maximum number of options that users can select, indicate it in a description.

If an option is recommended or the default, indicate it in its label.

Partially selected states should only be used when sub-options can be selected as children of a checkbox. A parent checkbox being in the partially selected state indicates that only some of its sub-options are selected.

---

## Related Components

If your use case doesn't match the guidelines above, consider using one of the following components instead:

-   [Multi select](#/form/MultiSelect) if there are more than seven options.
-   [Toggle](#/form/FlatSelect) if the result applies immediately, for example with a change of display.

> -   [Radio Button](#/form/RadioButton) if options are mutually exclusive.

Reviewer 2: ```suggestion
-   [Radio button](#/form/RadioButton) if options are mutually exclusive.
```

# Best Practices

> Use checkboxes to allow user to select multiple options or none of them.

Reviewer 2: ```suggestion
Use checkboxes to allow users to select one or more options from a list, or none at all.
```

> Use a stand-alone checkbox to toggle a single option on or off and when the effect is not immediate.

Reviewer 2: @kpoulincoveo Do we have an example for "when the effect is not immediate"? 🤔 
```suggestion
Use a stand-alone checkbox to toggle a single option on/off if the result is not immediately visible. Conversely, if the result applies immediately, for example a change of display, use a [Toggle](#/form/FlatSelect) instead.
```

> -   E.g. if a final confirmation is required to apply the effect.

Reviewer 2: For instance, you could use a checkbox to get final confirmation from a user before changes are applied.

Reviewer 2: Use a stand-alone checkbox to toggle a single option on/off if the result is not immediately visible, e.g., when a final confirmation from the user is required to apply changes.

Conversely, if the result is applied immediately, such as a change of display, use a [Toggle](#/form/FlatSelect) instead.

> Try to limit the number of available options under 7. If impossible, consider breaking them down into smaller logical sets or use the [Multi-Select](#/form/MultiSelect)

Reviewer 2: ```suggestion
Aim for seven or less options. If that's impossible, consider breaking your set of options into several smaller logical sets or using the [Multi select](#/form/MultiSelect) instead.
```

> Options should be ordered by most relevant unless a more suited ordering rational applies (such as for sizes or security levels). If no rational particularly stands-out, use alphanumerical order.

Reviewer 2: ```suggestion
Options should be sorted by most relevant unless a more suited ordering rationale applies, for example when presenting size or security level options. If no rationale stands out, place options in alphanumerical order.
```

> ## Labeling

Reviewer 2: Same question as in my comments on the RadioButtons PR: should we change this to `Text` since we have `Labels` below?

Author: I wouldn't because we have the _Help text and Instruction_ section, also that labeling section is meant to only refer to the Titles and options labels which are two type of "labels" that make your information architecture tangible and actionable for your users.
Let's keep labeling

Keep titles and labels short, preferably **under three words**.

### Title

> Always use a title if you have multiple options.

Reviewer 2: ```suggestion
Always use a title when offering multiple options.
```

> Use descriptive labels without action verbs.
> 
> -   E.g. Use "Grocery items" instead of "Select all items for your grocery".

Reviewer 2: ```suggestion
Use descriptive labels without action verbs. For example, write "Grocery items" rather than "Select all grocery items".
```

> For stand-alone checkbox, the title is optional. Use a title only if you need to dissociate the stand-alone checkbox from other adjacent options.

Reviewer 2: ```suggestion
For a stand-alone checkbox, the title is optional. Use a title only if you need to dissociate the stand-alone checkbox from other adjacent checkboxes.
```

### Labels

> Labels are used to name each option and should be self-explanatory as much as possible.

Reviewer 2: ```suggestion
Labels identify each option and should be self-explanatory.
```

Use a consistent writing style for all options in the list.

> For stand-alone checkbox, use a positive and active wording that clearly indicates what will happen if the checkbox is turned on.

Reviewer 2: ```suggestion
For a stand-alone checkbox, use a positive and active wording that clearly indicates what happens if the checkbox is selected.
```

> -   E.g. Use "Send me a text notification." instead of "Email notification".

Reviewer 2: ```suggestion
-   For example, write "Send me a text notification" rather than "Email notification".
```

> ## Help text and instructions

Reviewer 2: ```suggestion
## Help Text and Instructions
```

Instructions provide more context on the outcome of the choice at-hand. Use it sparingly, only when it is critical to the user's understanding.

> ## Feedback and validation

Reviewer 2: ```suggestion
## Feedback and Validation
```

> If you have a maximum number of options to select, add a description text to indicate it.

Reviewer 2: ```suggestion
If there is a maximum number of options that users can select, indicate it in a description.
```

> Indicate which options from the list are the default or recommended one in their labels

Reviewer 2: ```suggestion
If an option is recommended or the default, indicate it in its label.
```

> Partially selected states should only be used when sub-options can be selected as children of a checkbox. A parent checkbox being in the partially selected state indicates that only some of its sub-options are selected.

Reviewer 2: @kpoulincoveo I'm not sure what a partially selected state is? Can you show me what it looks like? I just want to be sure I understand this correctly before I make any changes 😅

Author: partial state is this one
![image](https://user-images.githubusercontent.com/44175746/152013084-6baf4d4c-a965-42c3-b566-e90cac3bb19f.png)

It's rarely used in the platform right now if I am not mistaken

Reviewer 2: ```suggestion
The partially selected state is only allowed with checkboxes that have children options. A partially selected parent checkbox indicates that only some of its children options are selected.
```

---

## Related Components

> If your use case doesn't match these guidelines, consider using an alternate component instead.

Reviewer 2: ```suggestion
If your use case doesn't match the guidelines above, consider using one of the following components instead:
```

> -   [Multi-Select](#/form/MultiSelectt) if you have more than 7 options.

Reviewer 2: ```suggestion
-   [Multi select](#/form/MultiSelect) if you have more than seven options.
```

> -   [Toggle](#/form/FlatSelect) if the result are instantaneous, like a change of display.

Reviewer 2: ```suggestion
-   [Toggle](#/form/FlatSelect) if the result applies immediately, for example with a change of display.
```
